### PR TITLE
glow: fix memory leak of old vertex data

### PR DESCRIPTION
**Pull request by michaelkirk**

### CODE REVIEW

File: ezgui/src/backend_glow.rs [hunk 1]
```diff
@@ -164,16 +165,87 @@ impl<'a> GfxCtxInnards<'a> {
...
+        unsafe {
+            gl.delete_vertex_array(self.id);
+        }
+    }
+}
+
+impl Drop for VertexArray {
+    fn drop(&mut self) {
+        assert!(
+            self.was_destroyed,
```
Comment (michaelkirk) [inline ezgui/src/backend_glow.rs:211]: I did this RIAA thing for the main OpenGL resources we're cycling through. It's a little heavier than strictly necessary, but I like being able to have this check here.

If it's controversial we could just do the simplest and inline the gl.destory calls in `Drawable#drop` and get rid of these structs.

File: ezgui/src/backend_glow.rs [hunk 2]
```diff
@@ -164,16 +165,87 @@ impl<'a> GfxCtxInnards<'a> {
...
+    vert_buffer: Buffer,
+    elem_buffer: Buffer,
     num_indices: i32,
+    gl: Rc<glow::Context>,
+}
+
+impl Drop for Drawable {
+    #[inline]
+    fn drop(&mut self) {
+        self.elem_buffer.destroy(&self.gl);
```
Comment (michaelkirk) [inline ezgui/src/backend_glow.rs:179]: Alternatively we could have an Rc<glow::Context> on `Buffer` and `VertexArray`, but this gives us a little Rc'ing.

**Comment by michaelkirk** [comment]
If this approach looks good, I can follow up with a PR to apply it to the glow/wasm backend (possibly refactoring to share more common code between the two).

File: ezgui/src/backend_glow.rs (hunk 2 above)
Comment (dabreegster) [reply ezgui/src/backend_glow.rs:179]: Having it on `Drawable` seems preferable; the 2 buffers and array are also a trio, so treating them as one unit that has to be destroyed and holds onto the context makes sense

**Comment by dabreegster** [comment]
Works fine for me! I was curious how quickly the memory leaked before, but I couldn't find a tool that'd show me Intel GPU memory usage on Linux.

Next steps of consolidating with the wasm impl sounds fine to me. Thanks for the PR!

**Comment by michaelkirk** [comment]
As a crude metric, for me on mac, running the montrose example at max speed in dev would balloon to 1GB within 30s.

Now it seems to hover around 170MB indefinitely.

**Comment by dabreegster** [comment]
Awesome, thanks for confirming! I made an attempt with the `total_bytes_uploaded` stuff to track this at the ezgui layer, but I've never been able to do external comparisons asking the OS.

**Pull request merged**